subsys/bluetooth/host/classic/sdp.c

```diff
@@ -1727,13 +1905,204 @@ static void sdp_client_notify_result(struct bt_sdp_client *session,
 	}
 }
 
+static int sdp_client_receive_ss(struct bt_sdp_client *session, struct net_buf *buf)
+{
+	struct bt_sdp_pdu_cstate *cstate;
+	uint16_t current_count;
+	uint16_t total_count;
+	uint32_t record_len;
+	uint32_t received_count;
+
+	/* Check the buffer len for the total_count field */
+	if (buf->len < sizeof(total_count)) {
+		LOG_ERR("Invalid frame payload length");
+		return 0;
+	}
+
+	/* Get total service record count. */
+	total_count = net_buf_pull_be16(buf);
+
+	/* Check the buffer len for the current_count field */
+	if (buf->len < sizeof(current_count)) {
+		LOG_ERR("Invalid frame payload length");
+		return 0;
+	}
+
+	/* Get current service record count. */
+	current_count = net_buf_pull_be16(buf);
+	/* Check valid of current service record count */
+	if (current_count > total_count) {
+		LOG_ERR("Invalid current service record count");
+		return 0;
+	}
+
+	received_count = session->rec_buf->len / SDP_RECORD_HANDLE_SIZE;
+	if ((received_count + current_count) > total_count) {
+		LOG_ERR("Excess data received");
+		return 0;
+	}
+
+	record_len = current_count * SDP_RECORD_HANDLE_SIZE;
+	if (record_len >= buf->len) {
+		LOG_ERR("Invalid packet");
+		return 0;
+	}
+
+	/* Get PDU continuation state */
+	cstate = (struct bt_sdp_pdu_cstate *)(buf->data + record_len);
+
+	if (cstate->length > BT_SDP_MAX_PDU_CSTATE_LEN) {
+		LOG_ERR("Invalid SDP PDU Continuation State length %u", cstate->length);
+		return 0;
+	}
+
+	if ((record_len + SDP_CONT_STATE_LEN_SIZE + cstate->length) >
+		buf->len) {
+		LOG_ERR("Invalid payload length");
+		return 0;
+	}
+
+	/*
+	 * No record found for given UUID. The check catches case when
+	 * current response frame has Continuation State shortest and
+	 * valid and this is the first response frame as well.
+	 */
+	if (!current_count && cstate->length == 0U &&
+		session->cstate.length == 0U) {
```

Comment:
I think these can also be merged to a single line, without violating the 100 max line length. That said, whenever you do have to split the lines, please align the split line so that it starts right after the opening parenthesis of the previous line (N tabs + 0-7 spaces)

Reply:
Yes. It done.